### PR TITLE
Allow Prow GKE Service Accounts to assume AWS roles

### PR DESCRIPTION
**Pull request by upodroid**
/cc @xmudrii @ameukam 

This is the AWS piece of getting Prow to run pods on EKS. The GKE bit is to adjust the manifests to have the correct env and a projected service account token with the correct claims.

/sig testing
/sig k8s-infra

### CODE REVIEW

**Comment by xmudrii** [comment]
/hold
some additional changes might be needed

File: infra/aws/terraform/prow-build-cluster/prow.tf
Comment (xmudrii) [inline infra/aws/terraform/prow-build-cluster/prow.tf:30]: Should we call this `Prow-EKS-Admin`?

Comment (upodroid) [reply infra/aws/terraform/prow-build-cluster/prow.tf]: Fixed

File: infra/aws/terraform/prow-build-cluster/prow.tf
Comment (upodroid) [inline infra/aws/terraform/prow-build-cluster/prow.tf:44]: @kubernetes/test-infra-maintainers Are these the right service accounts for pods that interact with kubernetes clusters?

File: infra/aws/terraform/prow-build-cluster/eks.tf
Comment (xmudrii) [inline infra/aws/terraform/prow-build-cluster/eks.tf:37]: Where this value comes from? Should that eventually be name of IAM role or that's something completely else? I'm still trying to understand how this ConfigMap works at all.

File: infra/aws/terraform/prow-build-cluster/prow.tf
Comment (xmudrii) [inline infra/aws/terraform/prow-build-cluster/prow.tf:24]: Maybe leave a comment where this value comes from (if not sensitive).

File: infra/aws/terraform/prow-build-cluster/prow.tf
Comment (xmudrii) [inline infra/aws/terraform/prow-build-cluster/prow.tf:27]: This comments needs an update. :)

File: infra/aws/terraform/prow-build-cluster/prow.tf
Comment (xmudrii) [inline infra/aws/terraform/prow-build-cluster/prow.tf:18]: Very tiny nitpick:
```suggestion
# This IAM configuration allows Prow GKE Clusters to assume a role on AWS
```

File: infra/aws/terraform/prow-build-cluster/eks.tf
Comment (upodroid) [reply infra/aws/terraform/prow-build-cluster/eks.tf:37]: This is an arbitary string. https://github.com/kubernetes-sigs/aws-iam-authenticator#crd-alpha

File: infra/aws/terraform/prow-build-cluster/prow.tf
Comment (upodroid) [reply infra/aws/terraform/prow-build-cluster/prow.tf:24]: No, it is the thumprint for *.googleapis.com

https://awscli.amazonaws.com/v2/documentation/api/latest/reference/iam/update-open-id-connect-provider-thumbprint.html
https://cloud.google.com/anthos/clusters/docs/multi-cloud/aws/how-to/use-workload-identity-aws

File: infra/aws/terraform/prow-build-cluster/prow.tf
Comment (upodroid) [reply infra/aws/terraform/prow-build-cluster/prow.tf:18]: Fixed

File: infra/aws/terraform/prow-build-cluster/prow.tf
Comment (TerryHowe) [reply infra/aws/terraform/prow-build-cluster/prow.tf:24]: Ah, trusted cert, but this is OIDC

File: infra/aws/terraform/prow-build-cluster/prow.tf
Comment (xmudrii) [inline infra/aws/terraform/prow-build-cluster/prow.tf:48]: This is missing comma.
```suggestion
                "system:serviceaccount:default:prow-controller-manager",
```

Comment (upodroid) [reply infra/aws/terraform/prow-build-cluster/prow.tf]: Fixed 

/label tide/merge-method-squash

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

The change is reconciled.

**Pull request merged**
